database/CoreDatabase.java

```diff
@@ -495,15 +492,15 @@ public void delete() {
     public static class IsolationManager {
 
         private final ConcurrentMap<CoreTransaction.Data, CommitState> commitStates;
-        private final ConcurrentNavigableMap<Long, Set<CoreTransaction.Data>> commitTimeline;
+        private final AtomicLong approximateLastCommitSnapshot;
         private final AtomicBoolean cleanupRunning;
 
         private enum CommitState {UNCOMMITTED, COMMITTING, COMMITTED}
 
         IsolationManager() {
-            this.cleanupRunning = new AtomicBoolean(false);
-            this.commitStates = new ConcurrentHashMap<>();
-            this.commitTimeline = new ConcurrentSkipListMap<>();
+            cleanupRunning = new AtomicBoolean(false);
+            commitStates = new ConcurrentHashMap<>();
+            approximateLastCommitSnapshot = new AtomicLong(0);
```

Comment:
For cleanup, we either need to know the last uncommitted transaction that means we can delete everything before that, OR we need to know the last commit snapshot that we can delete up to (because at any point we can open a new transaction that cannot be cleaned up from the isolation manager)